duckduckgo/macos-browser
# Address Bar/Search Experience improvements

Task/Issue URL: https://app.asana.com/0/1199237043630360/1200180252402997
Design URL: https://www.figma.com/file/GMASm53B24PF9R0N8MpoDi/Bookmarks-in-Search-Suggestions?node-id=1%3A17757
CC: @tomasstrba @samsymons @brindy 

**Description**:
This PR contains the following changes:
- First suggested URL (or bookmark) is autocompleted to the Address Bar
- To bring the URL results to the top, sorting is applied to the Suggestions
- Favorited Bookmark icon added for Suggestions

**Steps to test this PR**:
1. Begin typing a URL/Bookmark in the Address Bar
2. Ensure the address is autocompleted and search suggestion is higlighted
3. Ensure if the URL does not match the entered text it won't autocomplete
4. Ensure arrows/mouse navigation is working correctly
5. Ensure when backspace/pasteboard/input in the middle is performed then address is not autocompleted



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
